test/functional/test-custom-element.js

```diff
@@ -1012,88 +1011,110 @@ describes.realWin('CustomElement', {amp: true}, env => {
       }).to.throw(/Must never be called in template/); });
     });
 
+    // TODO(alabiaga): Fixes #19752. The window.matchMedia call in the code
+    // path of element.applySizesAndMediaQuery is not behaving consistently
+    // in headless mode, thus we mock the calls here. This is fine as we are
+    // not testing window behavior.
+    describe('apply sizes and media query', () => {
+      let element1;
+      let element2;
+      let matchMedia;
 
-    // TODO(alabiaga, 19752): flaky on Chrome 71
-    it.skip('should apply media condition', () => {
-      const element1 = new ElementClass();
-      element1.setAttribute('media', '(min-width: 1px)');
-      element1.applySizesAndMediaQuery();
-      expect(element1).to.not.have.class('i-amphtml-hidden-by-media-query');
+      beforeEach(() => {
+        element1 = new ElementClass();
+        matchMedia =
+          sandbox.stub(element1.ownerDocument.defaultView, 'matchMedia');
+        matchMedia.withArgs('(min-width: 1px)').returns({matches: true});
 
-      const element2 = new ElementClass();
-      element2.setAttribute('media', '(min-width: 1111111px)');
-      element2.applySizesAndMediaQuery();
-      expect(element2).to.have.class('i-amphtml-hidden-by-media-query');
-    });
+        element2 = new ElementClass();
+      });
 
-    // TODO(alabiaga, 19752): flaky on Chrome 71
-    it.skip('should apply sizes condition', () => {
-      const element1 = new ElementClass();
-      element1.setAttribute('sizes', '(min-width: 1px) 200px, 50vw');
-      element1.applySizesAndMediaQuery();
-      expect(element1.style.width).to.equal('200px');
+      it('should apply media condition', () => {
+        element1.setAttribute('media', '(min-width: 1px)');
+        element1.applySizesAndMediaQuery();
+        expect(element1).to.not.have.class('i-amphtml-hidden-by-media-query');
 
-      const element2 = new ElementClass();
-      element2.setAttribute('sizes', '(min-width: 1111111px) 200px, 50vw');
-      element2.applySizesAndMediaQuery();
-      expect(element2.style.width).to.equal('50vw');
-    });
+        matchMedia.restore();
 
-    // TODO(alabiaga, 19752): flaky on Chrome 71
-    it.skip('should apply heights condition', () => {
-      const element1 = new ElementClass();
-      element1.sizerElement = doc.createElement('div');
-      element1.setAttribute('layout', 'responsive');
-      element1.setAttribute('width', '200px');
-      element1.setAttribute('height', '200px');
-      element1.setAttribute('heights', '(min-width: 1px) 99%, 1%');
-      container.appendChild(element1);
-      element1.applySizesAndMediaQuery();
-      expect(element1.sizerElement.style.paddingTop).to.equal('99%');
+        matchMedia =
+           sandbox.stub(element2.ownerDocument.defaultView, 'matchMedia');
+        matchMedia.withArgs('(min-width: 1111111px)').returns({matches: false});
```

Comment:
Instead of restoring and stubbing again, use `withArgs()` to reuse the same stub but return different values for different args.

```js
    var callback = sinon.stub();
    callback.withArgs(42).returns(1);
    callback.withArgs(1).throws("name");
```

https://sinonjs.org/releases/latest/stubs/

Reply:
done. Thanks!